tests/components/motionblinds_ble/test_entity.py

```diff
@@ -0,0 +1,60 @@
+"""Tests for Motionblinds BLE entities."""
+
+from unittest.mock import patch
+
+import pytest
+
+from homeassistant.components.homeassistant import (
+    DOMAIN as HA_DOMAIN,
+    SERVICE_UPDATE_ENTITY,
+)
+from homeassistant.components.motionblinds_ble.const import (
+    ATTR_CONNECT,
+    ATTR_DISCONNECT,
+    ATTR_FAVORITE,
+    ATTR_SPEED,
+)
+from homeassistant.const import ATTR_ENTITY_ID, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry
+
+
+@pytest.mark.parametrize(
+    ("platform", "entity"),
+    [
+        (Platform.COVER, None),
+        (Platform.BUTTON, ATTR_CONNECT),
+        (Platform.BUTTON, ATTR_DISCONNECT),
+        (Platform.BUTTON, ATTR_FAVORITE),
+        (Platform.SELECT, ATTR_SPEED),
+    ],
+)
+async def test_entity_update(
```

Comment:
This is a platform concern, so this would rather fit a test in each platform

Reply:
That would only result in duplicate code? Why not just test it here that every entity responds to the update_entity service? I could also just move it to test_cover.py and only test the cover as that will still result in 100% coverage but this is better imo

Reply:
Hmm, the second thing is that we don't like `if` statements in the code because it makes it more unclear what its testing. So I would maybe prefer only the cover.

I am also low key wondering, why is this needed for the 100%? What is missing otherwise?

Reply:
> I am also low key wondering, why is this needed for the 100%? What is missing otherwise?

The `async_update` method of the MotionblindsBLEEntity class.

```python
    async def async_update(self) -> None:
        """Update state, called by HA if there is a poll interval and by the service homeassistant.update_entity."""
        _LOGGER.debug("(%s) Updating entity", self.entry.data[CONF_MAC_CODE])
        await self.device.status_query()
```

> Hmm, the second thing is that we don't like if statements in the code because it makes it more unclear what its testing. So I would maybe prefer only the cover.

In my opinion the added parametrize and condition are worth it to cover all platforms and make sure each of them can be used to update the device state (after all they are all an instance of MotionblindsBLEEntity), but if you insist on only doing it for the cover platform let me know and I will change it

Reply:
In my opinion, this test is OK and reduces duplicated code except for the cover variant. We should avoid if statements in tests, and therefore please split the cover variant of this test into a new test, placed in test_cover.py

Reply:
I do not see the appeal of duplicating this exact same code for the test_cover.py over using a simple if statement that determines the entity_id. I could also create a get_unique_id() method that constructs the unique_id based on the platform and type of entity such that the if statement is no longer in the test? For now though I have removed the cover platform as it will reach 100% coverage either way as long as one platform is tested